### DISCUSSION TITLE
How to avoid force type casting to include dynamically populated fields?

### DISCUSSION THREAD

**Discussion by hanza93** [Automattic/mongoose#14889] [General]
I'm trying to call a function that expects a `User` type, but I need it to work with dynamically populated fields. Here’s my setup:

**User Schema:**
```ts
@Schema(schemaOptions)
export class User {
  @Prop({ type: MongooseSchema.Types.ObjectId, ref: ModelReferences.Profile })
  profile: Types.ObjectId;

  @Prop({ type: String })
  name: string;
}

export const UserSchema = SchemaFactory.createForClass(User);
UserSchema.loadClass(User);
```

**Profile Schema:**
```ts
@Schema(schemaOptions)
export class Profile {
  @Prop({ type: String })
  address: string;
}

export const ProfileSchema = SchemaFactory.createForClass(Profile);
ProfileSchema.loadClass(Profile);
```

I have a function that expects a `User` type:

```ts
export const getName = (user: User) => user.name;
```

When I call this function from another method where I populate the `Profile`, I run into a type issue:

```ts
export const async doSomething(userId: string) = {
  const user = await models.User.findOne({ _id: userId }).populate<{profile: Profile}>('profile');
  
  // Do some stuff with profile here
  const name = getName(user as unknown as User);
};
```

**Question:**  
Is there a way to make the `getName` function accept dynamically populated fields without forcing type casting? How can I ensure the `User` type is compatible with populated fields like `profile`?

**Comment by vkarpov15** [comment]
Does `export const getName = (user: User & { profile: Profile }) => user.name;` work for you?
